packages/devtools_app/test/memory/diff/controller/snapshot_item_test.dart

```diff
@@ -0,0 +1,36 @@
+// Copyright 2023 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:devtools_app/src/screens/memory/panes/diff/controller/snapshot_item.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+import '../../../test_infra/test_data/memory/heap/heap_data.dart';
+
+void main() {
+  test(
+    '$SnapshotDataItem serializes and deserializes correctly',
+    () async {
+      final item = SnapshotDataItem(
+        defaultName: 'defaultName',
+        displayNumber: 1,
+        nameOverride: 'nameOverride',
+      );
+      await item.loadHeap(HeapGraphLoaderGoldens());
+
+      final json = item.toJson();
```

Comment:
assert the values created by toJson to verify expectations

Reply:
IMHO, it will increase maintenance cost of the test, without adding value.
We do not care what is in json, we just want it to produce the same values when deserialized, right?

Reply:
If the `.toJson()` value is not what we expect (like a field is missing for example or a default value is wrong), bugs could still slip in. For example both `.toJson()` and `fromJson()` could be missing a field.

Reply:
I added validation that all keys are used. I skipped validation for values. 
Yes, bugs still can squeeze, but no test coverage is full. I believe verification for values is overhead in this case.